Update to Core SDK to fail fast during Pairing (#293)
WalletConnect/walletconnect-docs: docs/specs/clients/core/pairing/readme.md

#### Successful request handling if B registers protocol P

1. A creates random symKey S and pairing topic that is a sha256 hash of symKey S.

> 2. A's pairing client registers protocol P by passing the list of methods needed for protocol P

Reviewer: ```suggestion
2. A's pairing client registers protocol P by passing the list of methods required for protocol P
```

Just wording to match semantics

> 2. A's pairing client registers protocol P by passing the list of methods required for protocol P

Reviewer 2: all protocols P(p1,p2 etc.) are registered individually by it's sdks. we are not passing the list of methods

Author: removed

3. A encrypts request with symKey S. Pairing client does not expects any response. P is responsible for response subscription.
4. A sends request on pairing topic.

> 5. A generates URI and appends the query parameter with the keys "and-methods" and "or-methods" the value being a array.

Reviewer: ```suggestion
5. A generates URI and appends the query parameter with the keys "req-methods" and "opt-methods" the value being a array.
```

Looking at the example above you used "req-methods" and "opt-methods"

> 5. A generates URI and appends the query parameter with the keys "req-methods" and "opt-methods" the value being a array.

Reviewer 3: ```suggestion
5. A generates URI and appends the query parameter with the keys "req-methods" and "opt-methods", the value of each being an array.
```

> 6. B's pairing client registers protocol P by passing lists of methods needed for Protocol P

Reviewer 2: all protocols P(p1,p2 etc.) are registered individually by it's sdks. we are not passing the list of methods

Author: removed

> 7. B verifies that the URI's "and-methods" query paramter's value equals B's cached "and" list of protocol methods along with checking if B's cached "or" list matches any values in the URI's "or-methods' query parameter's value and then subscribes to pairing topic from provided URI

Reviewer: Let's stick to one semantics "and" vs "req"; "or" vs "opt"

> 7. B verifies that the URI's "req-methods" query paramter's value equals B's cached "and" list of protocol methods along with checking if B's cached "or" list matches any values in the URI's "opt-methods' query parameter's value and then subscribes to pairing topic from provided URI

Reviewer 3: Typos:

```suggestion
7. B verifies that the URI's "req-methods" query parameter's value equals B's cached "and" list of protocol methods along with checking if B's cached "or" list matches any values in the URI's "opt-methods" query parameter's value and then subscribes to pairing topic from provided URI
```

---

Also not sure if we still want the `AND/OR` terminology here

8. B receives protocol P request.

#### Error case where B did not registered protocol P

1. A creates random symKey S and pairing topic that is a sha256 hash of symKey S.

> 2. A's pairing client registers protocol P by passing the list of methods needed for protocol P

Reviewer 2: same here

3. A encrypts request with symKey S. Pairing client does not expects any response. P is responsible for response subscription.
4. A sends request on pairing topic.
5. A generates URI and appends the query parameter with a key "methods" with the value being a list of the required methods registered.

> 6. B's pairing client registers protocol P by passing the list of methods needed for Protocol P

Reviewer 2: same

> 7. B verifies that the URI's "req-methods" query paramter's value equals B's cached "and" list of protocol methods along with checking if B's cached "or" list matches any values in the URI's "opt-methods' query parameter's value and the verification fails by either not having all of the required methods or missing any of the optional methods. B will notify user that it cannot pair with A, and internally publish a wc_deletePairing with a reason.

Reviewer 3: Typos, sentence length:

```suggestion
7. B verifies that the URI's "req-methods" query parameter's value equals B's cached "and" list of protocol methods along with checking if B's cached "or" list matches any values in the URI's "opt-methods" query parameter's value. The verification fails by either not having all of the required methods or missing any of the optional methods. B will notify user that it cannot pair with A, and internally publish a wc_deletePairing with a reason.
```
